gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java

```diff
@@ -453,7 +453,9 @@ public boolean setIdentity(String privateKey, JSch jsch) {
   public long getFileMTime(String filePath) throws FileBasedHelperException {
     try {
       ChannelSftp channelSftp = getSftpChannel();
-      return channelSftp.lstat(filePath).getMTime();
+      int modificationTime = channelSftp.lstat(filePath).getMTime();
```

Comment:
Good catch!

Could you put the channelSftp.disconnect() in a try-finally block?